src/Directions.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+
+import { MapContext } from './Map'
+
+import { createCoordinate, NumberTuple } from './utils'
+
+type annotationOptions = {
+  origin?: mapkit.MarkerAnnotationConstructorOptions
+  destination?: mapkit.MarkerAnnotationConstructorOptions
+}
+
+type DirectionsProps = {
+  origin: NumberTuple
+  destination: NumberTuple
+  transportType?: 'Automobile' | 'Walking'
```

Comment:
I wonder if we can pull these out of the typescript vs hardcoding them.

Reply:
Are you suggesting including DirectionsRequest directly from Mapkit JS library as seen [here](https://developer.apple.com/documentation/mapkitjs/directionsrequest)

```suggestion
request: mapkit.DirectionRequest
```

Reply:
no, I was just thinking instead of having `'Automobile' | 'Walking'` here we could get the values from the mapkit types. As of now these are not provided however. I did just move the types into this package, so we could add these, but I'm not sure if its really worth it at this point.

Reply:
A way around this is to remove all three of these props and replace it with request as show within [the Apple Documentation](https://developer.apple.com/documentation/mapkitjs/directionsrequest). It appears that trying to set the transportType type to mapkit.Directions.Transport fails due to mapkit having no exports called Directions
```suggestion
  request: mapkit.DirectionsRequest
```

Reply:
Regarding my above comment, `mapkit.Directions.Transport` is just of type string, therefore setting this to type string would work, however users would still need to use `mapkit.Directions.Transport.Automobile` etc. Having default strings of `'AUTOMOBILE' | 'WALKING'` (just discovered they need to be capitalised may make this component simpler to use?

Reply:
> no, I was just thinking instead of having `'Automobile' | 'Walking'` here we could get the values from the mapkit types. As of now these are not provided however. I did just move the types into this package, so we could add these, but I'm not sure if its really worth it at this point.

Sorry your reply has only just come through! I see what you're saying. As a longer term it may be worth adding these, but I'm unsure currently whether more transport types are planned to be added or not, and therefore it is currently manageable with just the two types?